### DISCUSSION TITLE
Improvement on the export/report feature

### DISCUSSION THREAD

**Discussion by wei2go** [verbb/formie#773] [Ideas]
Our clients often need a more comprehensive reporting feature than what Formie currently has. It would be great if it can be improved, so in CP:
- We can select a date range for the export
- We can select which fields (columns) to be exported.
- The exported columns are field labels instead of the handles (or we can toggle showing field labels VS handles in settings).
- We can save the export profiles to remember the settings mentioned above.

**Comment by engram-design** [comment]
We do use Craft's element exporting, so most of this would be an add-on. Particularly selecting which fields to be exported, as adding date ranges is achievable (similar to Commerce) https://github.com/verbb/formie/issues/594. Exported column labels is a good point, and export profiles isn't something I've seen anywhere before (yet).

I'm keen to see what filtering Craft 4 (and Craft 5) brings, as it'd be great to keep things on-par with native Craft behaviour.

**Reply by wei2go** [reply]
Yeah among my dot points the date range and column labels is probably the most needed one. 

The others are not as important 
- Unneeded columns can be deleted via Excel or Google Sheets, etc. 
- The export profile is a nice-to-have.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.7](https://github.com/verbb/formie/releases/tag/4.0.0-beta.7) with a whole new Reports handling.
